R/build.R

```diff
@@ -27,19 +40,18 @@ inlineCxxPlugin <- function() {
    )
 }
 
-mtuneFlags <- function() {
-   switch(Sys.info()[["sysname"]],
-          "Linux"   = "-mtune=native",
-          "Darwin"  = "-mtune=core2",
-          "Windows" = "-mtune=core2",
-          ""
-   )
-}
-
-tbbCxxFlags <- function() {
-   flags <- "$(CXX1XSTD)"
+tbbCxxFlags <- function(simd = TRUE) {
```

Comment:
I'm not sure that works. There actually are no SIMD flags (since it's
enabled by default). For simd = FALSE we add -DRCPP_PARALLEL_USE_SIMD=0, so
the function would end up being called simdDisabledCxxFlags and be only
called from one place.

On Fri, Jan 15, 2016 at 1:32 PM, Kevin Ushey <notifications@github.com>
wrote:

> In R/build.R
> <https://github.com/RcppCore/RcppParallel/pull/21#discussion_r49887307>:
>
> > @@ -27,19 +40,18 @@ inlineCxxPlugin <- function() {
> >     )
> >  }
> >
> > -mtuneFlags <- function() {
> > -   switch(Sys.info()[["sysname"]],
> > -          "Linux"   = "-mtune=native",
> > -          "Darwin"  = "-mtune=core2",
> > -          "Windows" = "-mtune=core2",
> > -          ""
> > -   )
> > -}
> > -
> > -tbbCxxFlags <- function() {
> > -   flags <- "$(CXX1XSTD)"
> > +tbbCxxFlags <- function(simd = TRUE) {
>
> I wonder if the SIMD stuff should be in its own function, e.g.
> simdCxxFlags(), and then the cxxFlags() argument would be the paste()d
> output from each?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RcppCore/RcppParallel/pull/21/files#r49887307>.
>